solr/solrj/src/test/org/apache/solr/client/solrj/cloud/autoscaling/TestPolicy.java

```diff
@@ -1709,13 +1708,134 @@ public DistribStateManager getDistribStateManager() {
     assertEquals(2, s1.getRefCount());
 
     s2[0].release();
-    assertFalse(sessionRef.getSessionWrapper() == PolicyHelper.SessionWrapper.DEFAULT_INSTANCE);
+    assertFalse(sessionRef.isEmpty());
     s1.release();
-    assertTrue(sessionRef.getSessionWrapper() == PolicyHelper.SessionWrapper.DEFAULT_INSTANCE);
+    assertTrue(sessionRef.isEmpty());
 
 
   }
 
+  @Test
```

Comment:
Might want to add a test for creationsInProgress? But would probably require refactoring/exposing creationsInProgress

Reply:
Yes. Not sure it's worth it. We'll see what others think.